File: U2-Toolkit/.NET Samples/samples/C#/UniData/EntityFramework_CodeFirst_MultiValue1/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Data.Entity;
using System.ComponentModel.DataAnnotations;
using System.Data.Entity.ModelConfiguration.Conventions;

namespace EntityFramework_CodeFirst_MultiValue1
{
    [Table("STUDENT")]
    public class Student
    {
        Student()
        {
        }

        [Key]
        [Column("ID")]
        public string StudentID { get; set; }  //Single Value
        [Column("FNAME")]
        public string FirstName { get; set; } //Single Value
        [Column("LNAME")]
        public string LastName { get; set; } //Single Value
        [Column("SEMESTER")]
        public string Semester { get; set; } //Multi Value

    }
    public class StudentContext : DbContext
    {
        public StudentContext()
        {
        }

        public DbSet<Student> Students { get; set; }

        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            base.OnModelCreating(modelBuilder);
            modelBuilder.Conventions.Remove<PluralizingTableNameConvention>();
        }
    }
}

File: U2-Toolkit/.NET Samples/samples/VB.NET/UniData/EntityFramework_CodeFirst_MultiValue1/Student.vb
﻿Imports System.Data.Entity
Imports System.ComponentModel.DataAnnotations
Imports System.Data.Entity.ModelConfiguration.Conventions

Namespace EntityFramework_CodeFirst_MultiValue1
    <Table("STUDENT")> _
    Public Class Student
        Sub New()
        End Sub

        'Single Value
        <Key()> _
        <Column("ID")> _
        Public Property StudentID() As String
            Get
                Return m_StudentID
            End Get
            Set(value As String)
                m_StudentID = Value
            End Set
        End Property
        Private m_StudentID As String

        'Single Value
        <Column("FNAME")> _
        Public Property FirstName() As String
            Get
                Return m_FirstName
            End Get
            Set(value As String)
                m_FirstName = Value
            End Set
        End Property
        Private m_FirstName As String

        'Single Value
        <Column("LNAME")> _
        Public Property LastName() As String
            Get
                Return m_LastName
            End Get
            Set(value As String)
                m_LastName = Value
            End Set
        End Property
        Private m_LastName As String

        'Multi Value
        <Column("SEMESTER")> _
        Public Property Semester() As String
            Get
                Return m_Semester
            End Get
            Set(value As String)
                m_Semester = Value
            End Set
        End Property
        Private m_Semester As String

    End Class
    Public Class StudentContext
        Inherits DbContext
        Public Sub New()
        End Sub

        Public Property Students() As DbSet(Of Student)
            Get
                Return m_Students
            End Get
            Set(value As DbSet(Of Student))
                m_Students = Value
            End Set
        End Property
        Private m_Students As DbSet(Of Student)

        Protected Overrides Sub OnModelCreating(modelBuilder As DbModelBuilder)
            MyBase.OnModelCreating(modelBuilder)
            modelBuilder.Conventions.Remove(Of PluralizingTableNameConvention)()
        End Sub
    End Class
End Namespace
